4.6.2/Dockerfile

```diff
@@ -9,6 +9,27 @@ RUN Add-WindowsFeature Web-Server; \
 
 ADD ServiceMonitor.exe /
 
+RUN \
+	#download and unzip Roslyn nupkg
+	Invoke-WebRequest  https://api.nuget.org/packages/microsoft.net.compilers.2.3.1.nupkg -OutFile c:\microsoft.net.compilers.2.3.1.zip ; \	
+	Expand-Archive -Path c:\microsoft.net.compilers.2.3.1.zip -DestinationPath c:\RoslynCompilers ; \
+	Remove-Item c:\microsoft.net.compilers.2.3.1.zip -Force ; \
+
+	#ngen Roslyn binaries
+	& 'C:\Windows\Microsoft.NET\Framework64\v4.0.30319\ngen.exe' install c:\RoslynCompilers\tools\csc.exe /ExeConfig:c:\RoslynCompilers\tools\csc.exe ; \
```

Comment:
This is really a nested foreach (over bitness and EXE). It could be written something like:
```
$roslynToolsPath = 'C:\RoslynCompilers\tools'
foreach($framework in (Join-Path "C:\Windows\Microsoft.NET" "Framework*" -Resolve)) {
  $ngen = Join-Path $framework 'v4.0.30319\ngen.exe'
  foreach($binary in @('csc.exe', 'vbc.exe', 'VBCSCompiler.exe')) {
    $exePath = Join-Path $roslynToolsPath $binary
    & $ngen install $exePath /ExeConfig:$exePath
    if ($LASTEXITCODE -ne 0){ Exit $LASTEXITCODE }
  }
}
```

It's more compact, but do you think that would be clearer?

Reply:
I tend to agree that the current pattern is more readable and the pattern you'd typically see within a Dockerfile.

Reply:
Wish there was a way to avoid needing the check the LASTEXITCODE on every command.  Would be nice if you could cleanly && the results and let RUN handle non-zero results.  I am not familiar enough with PS to answer this.